sdk/daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/interpretation/Error.scala

```diff
@@ -148,13 +148,17 @@ object Error {
 
     sealed abstract class Error extends Serializable with Product
 
-    // TODO: https://github.com/DACH-NY/canton-network-utilities/issues/2916: introduce structured CCTP error reporting
-    //  sealed case class CCTP(error: CCTP.Error) extends Error
-    //  object CCTP {
-    //    sealed abstract class Error extends Serializable with Product
-    //    final case class InvalidKeyError(msg: String) extends Error
-    //    final case class SignatureError(msg: String) extends Error
-    //  }
+    sealed case class CCTP(error: CCTP.Error) extends Error
+
+    object CCTP {
+      sealed abstract class Error extends Serializable with Product
+
+      final case class InvalidByteEncoding(value: String, cause: String) extends Error
+
+      final case class InvalidKeyError(msg: String) extends Error
+
+      final case class SignatureError(msg: String) extends Error
```

Comment:
Rename using `Malformed` as base.

specify key and signature as error arguments